src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -173,6 +174,36 @@ public function loadClass($class)
                 @trigger_error(sprintf('The %s class is considered final%s. It may change without further notice as of its next major version. You should not extend it from %s.', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
             }
 
+            // Only check final methods if the class has a parent
+            if ($parent) {
+                foreach ($refl->getMethods(\ReflectionMethod::IS_PUBLIC | \ReflectionMethod::IS_PROTECTED) as $method) {
+                    try {
+                        $prototype = $method->getPrototype();
```

Comment:
Nice finding.
Then I'm wondering if we need it at all :)
Parent classes are always loaded before their children. This means each class should only look at methods that has `$method->class === $class` (let's use properties as much as possible, performance is critical here).
By populating the `self::$finalMethods` props, then child should have everything to decide quickly.